packages/react-charting/src/components/GaugeChart/GaugeChart.base.tsx

```diff
@@ -0,0 +1,627 @@
+import * as React from 'react';
+import * as shape from 'd3-shape';
+import { classNamesFunction, getRTL } from '@fluentui/react/lib/Utilities';
+import {
+  IGaugeChartProps,
+  IGaugeChartSegment,
+  IGaugeChartStyleProps,
+  IGaugeChartStyles,
+  GaugeValueFormat,
+} from './GaugeChart.types';
+import { IProcessedStyleSet } from '@fluentui/react/lib/Styling';
+import {
+  Points,
+  convertToLocaleString,
+  formatValueWithSIPrefix,
+  getAccessibleDataObject,
+  getColorFromToken,
+  pointTypes,
+} from '../../utilities/index';
+import { ILegend, LegendShape, Legends, Shape } from '../Legends/index';
+import { FocusZone, FocusZoneDirection } from '@fluentui/react-focus';
+import { Callout, DirectionalHint } from '@fluentui/react/lib/Callout';
+import { IYValueHover } from '../../index';
+import { TooltipText } from '../../utilities/TooltipText';
+import { select as d3Select } from 'd3-selection';
+
+const GAUGE_MARGIN = 16;
+const LABEL_WIDTH = 36;
+const LABEL_HEIGHT = 16;
+const LABEL_OFFSET = 4;
+const TITLE_OFFSET = 11;
+const EXTRA_NEEDLE_LENGTH = 4;
+export const ARC_PADDING = 2;
+export const BREAKPOINTS = [52, 70, 88, 106, 124, 142];
+const ARC_WIDTHS = [12, 16, 20, 24, 28, 32];
+export const FONT_SIZES = [20, 24, 32, 32, 40, 40];
+
+const getClassNames = classNamesFunction<IGaugeChartStyleProps, IGaugeChartStyles>();
+
+interface IYValue extends Omit<IYValueHover, 'y'> {
+  y?: string | number;
+}
+export interface IGaugeChartState {
+  hoveredLegend: string;
+  selectedLegend: string;
+  focusedElement: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  calloutTarget: any;
+  isCalloutVisible: boolean;
+  hoverXValue: string | number;
```

Comment:
A gauge chart will have a single value. Then what does hover x and hover y mean?

Reply:
The hoverXValue and hoverYValues states are used to render all legends with their ranges at the bottom of hover card.